src/filterA.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @since TODO
+ */
+import { HKT, Kind, Kind2, Kind3, Kind4, URIS, URIS2, URIS3, URIS4 } from 'fp-ts/lib/HKT'
+import { Applicative, Applicative1, Applicative2, Applicative3, Applicative4 } from 'fp-ts/lib/Applicative'
+import * as A from 'fp-ts/lib/Array'
+import * as O from 'fp-ts/lib/Option'
+
+/**
+ * TODO describe here
```

Comment:
you can use the Haskell [`filterM`](https://hackage.haskell.org/package/base-4.12.0.0/docs/src/Control.Monad.html#filterM) description here

> This generalizes the array-based 'filter' function